modules/BiomesAPI/src/main/java/org/terasology/biomesAPI/BiomeManager.java

```diff
@@ -149,6 +150,10 @@ public void checkBiomeChangeEvent(MovedEvent event, EntityRef entity) {
 
             Biome newBiome = newBiomeOptional.get();
             Biome oldBiome = oldBiomeOptional.get();
+            if (movedFlag == false) {
```

Comment:
I don't really like this approach. Could you maybe catch something like PlayerSpawnEvent, if it would be sensible?